From 2bf3fffe938e5206d4841e020743adfab10d98ab Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 31 Dec 2021 09:34:44 +0100
Subject: [PATCH] Dont't add SDKROOT to vals

---
 tool/mkconfig.rb | 2 --
 1 file changed, 2 deletions(-)

diff --git a/tool/mkconfig.rb b/tool/mkconfig.rb
index e6d61c5..3327d61 100755
--- a/tool/mkconfig.rb
+++ b/tool/mkconfig.rb
@@ -132,8 +132,6 @@ File.foreach "config.status" do |line|
       val = '"$(target_cpu)-$(target_vendor)-$(target_os)"'
     when /^host(?:_(?:os|vendor|cpu|alias))?$/
       val = %["$(#{name.sub(/^host/, 'target')})"]
-    when /^includedir$/
-      val = '"$(SDKROOT)"'+val if /darwin/ =~ arch
     end
     v = "  CONFIG[\"#{name}\"] #{eq} #{val}\n"
     if fast[name]
--
2.40.0
