From 3228c48129d596d5a062aa7579d499ba561119c4 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 31 Dec 2021 09:34:44 +0100
Subject: [PATCH] Dont't add SDKROOT to vals

---
 tool/mkconfig.rb | 2 --
 1 file changed, 2 deletions(-)

diff --git a/tool/mkconfig.rb b/tool/mkconfig.rb
index 6e23af5..ad84b21 100755
--- a/tool/mkconfig.rb
+++ b/tool/mkconfig.rb
@@ -124,8 +124,6 @@ File.foreach "config.status" do |line|
       if universal
         platform = val.sub(/universal/, %q[#{arch && universal[/(?:\A|\s)#{Regexp.quote(arch)}=(\S+)/, 1] || RUBY_PLATFORM[/\A[^-]*/]}])
       end
-    when /^includedir$/
-      val = '"$(SDKROOT)"'+val if /darwin/ =~ arch
     end
     v = "  CONFIG[\"#{name}\"] #{eq} #{val}\n"
     if fast[name]
-- 
2.34.1

