diff --git a/vm_dump.c b/vm_dump.c
index 578f5c0..26f6355 100644
--- a/vm_dump.c
+++ b/vm_dump.c
@@ -713,12 +713,13 @@ rb_print_backtrace(void)
 #define MAX_NATIVE_TRACE 1024
     static void *trace[MAX_NATIVE_TRACE];
     int n = (int)backtrace(trace, MAX_NATIVE_TRACE);
-#if (defined(USE_ELF) || defined(HAVE_MACH_O_LOADER_H)) && defined(HAVE_DLADDR) && !defined(__sparc)
+#if FALSE
     rb_dump_backtrace_with_lines(n, trace);
 #else
     char **syms = backtrace_symbols(trace, n);
     if (syms) {
 	int i;
+        fprintf(stderr, "conda-forge::ruby can't rb_dump_backtrace_with_lines\n");
 	for (i=0; i<n; i++) {
 	    fprintf(stderr, "%s\n", syms[i]);
 	}
